client/pkg/srv/srv.go

```diff
@@ -106,9 +106,10 @@ func GetClient(service, domain string, serviceName string) (*SRVClients, error)
 			return err
 		}
 		for _, srv := range addrs {
+			shortHost := strings.TrimSuffix(srv.Target, ".")
```

Comment:
I thought the trailing dot meant the dns name was canonical and shouldn't have dns search paths appended to it. Does trimming the dot mean local dns search paths are used and the resolved address could end up being different?

Reply:
If you keep reading the following [couple of lines of code](https://github.com/etcd-io/etcd/blob/ff1569f1340b2299204aa944304ee344ffe486f7/client/pkg/srv/srv.go#L109-L112), you will find we just trim the trailing dot when constructing the network address.

Reply:
if the SRV records are being returned with trailing dots, and we're turning that into client URLs without trailing dots, doesn't that mean DNS search paths are used to resolve those URLs to IPs, contrary to what the trailing dot in the srv record indicated?

Reply:
The `addrs` is already the results of dns lookup. The [urls/endpoints](https://github.com/etcd-io/etcd/blob/ff1569f1340b2299204aa944304ee344ffe486f7/client/pkg/srv/srv.go#L110) will be delivered to [gRPC](google.golang.org/grpc)

Reply:
won't a second DNS lookup be done to resolve the hostnames to IPs? don't we still need the trailing `.` to make *that* DNS resolution avoid using locally configured DNS search paths?

Reply:
I verified this change is incorrect, opened https://github.com/etcd-io/etcd/issues/13948 with details of the implications for DNS lookup.

Reply:
revert opened in https://github.com/etcd-io/etcd/pull/13949

Reply:
that also probably might mean the server-side SRV record transformation this was based on is incorrect - https://github.com/etcd-io/etcd/blob/e814f6f78a1213a344038447cd1e48949327b36c/client/pkg/srv/srv.go#L72 is probably worth revisiting

Reply:
Please see my response in [13948#issuecomment-1100588121)](https://github.com/etcd-io/etcd/issues/13948#issuecomment-1100588121)